instrumentation/resque/lib/opentelemetry/instrumentation/resque/patches/resque_module.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Resque
+      module Patches
+        # Module to prepend to Resque for instrumentation
+        module ResqueModule
+          def self.prepended(base)
+            class << base
+              prepend ClassMethods
+            end
+          end
+
+          # Module to prepend to Resque singleton class
+          module ClassMethods
+            def push(queue, item)
+              # Check if the job is wrapped is being wrapped by ActiveJob
+              # before retrieving the job class name
+              job_class = if item[:args][0]&.is_a?(Hash)
+                            item[:args][0]['job_class']
+                          else
+                            item[:class]
+                          end
```

Comment:
Is there any way to do this more reliably or do we have to implicitly depend on AJ internals here?

Reply:
I don't think so.  I don't think this is particularly great either but I'm not sure what the better option is to maintain compatibility.  

It's how the resque UI itself does it https://github.com/resque/resque/pull/1525/files#diff-c63425703f7922acdeace3db5e0c9d98fb6ab3ae04fdab863c7161b46bb3d618R2

Reply:
They do have the extra check there for:
```ruby
job['class'] == 'ActiveJob::QueueAdapters::ResqueAdapter::JobWrapper'
```
Is that worth adding?

Reply:
> Is that worth adding?

👍 I think so

> Is there any way to do this more reliably or do we have to implicitly depend on AJ internals here?

It is ugly, but I also think this is about the best way to do it. 😞

Reply:
Updated https://github.com/open-telemetry/opentelemetry-ruby/pull/779/commits/26e24198f99d8ec739bc9862ade9644997f1f796